pkgs/development/compilers/rust/print-hashes.sh

```diff
@@ -1,8 +1,21 @@
-#!/bin/sh
+#!/usr/bin/env bash
+
+# All rust-related downloads can be found at
+# https://static.rust-lang.org/dist/index.html.  To find the date on
+# which a particular thing was last updated, look for the *-date.txt
+# file, e.g.
+# https://static.rust-lang.org/dist/channel-rust-beta-date.txt
 
 PLATFORMS="i686-unknown-linux-gnu x86_64-unknown-linux-gnu i686-apple-darwin x86_64-apple-darwin"
 BASEURL="https://static.rust-lang.org/dist"
-VERSION=$1
+DATE=$1
+VERSION=$2
+
+if [[ -z  $DATE ]]
+then
+    echo "No date supplied"
+    exit -1
+fi
```

Comment:
This doesn't appear to work:

```
grahamc@Morbo> ./print-hashes.sh 2017-06-08 1.17.0
<?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>dist/2017-06-08/rust-1.17.0-i686-unknown-linux-gnu.tar.gz.sha256</Key><RequestId>C34D6395F7683D89</RequestId><HostId>lr90iQsHnNluvVulVA4Nbf0S96ZYksq0aGQeZ3FBKO+L2ATnW7SV7ujZnG2rOzENjTTzvy5LhHY=</HostId></Error><?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>dist/2017-06-08/rust-1.17.0-x86_64-unknown-linux-gnu.tar.gz.sha256</Key><RequestId>33D783FED92A3750</RequestId><HostId>cr7TZsC9Xe29GI9sCWJhSxc0FpXboooNOwZHln0ZsBKnO5djQEQG5bP7uNDvKaIE438Dxl9Cw44=</HostId></Error><?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>dist/2017-06-08/rust-1.17.0-i686-apple-darwin.tar.gz.sha256</Key><RequestId>236CB3DD279BECAE</RequestId><HostId>xyplFt5QbqswtiSOQxH8mUpLBLKuBC+3T92kKD7V+9KqjKKqYbB1Y+kHDycz+dNu6zeQ5hTTd2w=</HostId></Error><?xml version="1.0" encoding="UTF-8"?>
<Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Key>dist/2017-06-08/rust-1.17.0-x86_64-apple-darwin.tar.gz.sha256</Key><RequestId>6591E6D2614E1664</RequestId><HostId>fimagRKeC4GecGI/YFO0MDvtaUibYKh3vlFY7eIvoKlTH5/LVxk949K1ryXBGixiMo7d0NSwpwk=</HostId></Error>%                                                                                                                  
```

when using https://static.rust-lang.org/dist/channel-rust-stable-date.txt